app/src/main/java/com/android/sample/ui/preferences/PreferencesScreen.kt

```diff
@@ -0,0 +1,156 @@
+package com.android.sample.ui.preferences
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.ArrowBack
+import androidx.compose.material3.*
+import androidx.compose.runtime.*
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.model.preferences.Preferences
+import com.android.sample.model.preferences.PreferencesViewModel
+import com.android.sample.model.preferences.UnitsSystem
+import com.android.sample.model.preferences.WeightUnit
+import com.android.sample.ui.composables.SaveButton
+import com.android.sample.ui.navigation.NavigationActions
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun PreferencesScreen(
+    navigationActions: NavigationActions,
+    preferencesViewModel: PreferencesViewModel
+) {
+  val preferences =
+      preferencesViewModel.preferences.collectAsState().value
+          ?: return Text(text = "No Preferences registred. Should not happen", color = Color.Red)
+
+  var unitsSystem by remember { mutableStateOf(preferences.unity) }
+  var weightUnit by remember { mutableStateOf(preferences.weight) }
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = { Text("Modify preferences", fontSize = 20.sp, fontWeight = FontWeight.Bold) },
+            navigationIcon = {
+              IconButton(onClick = { navigationActions.goBack() }) {
+                Icon(imageVector = Icons.Default.ArrowBack, contentDescription = "Back")
+              }
+            },
+            colors = TopAppBarDefaults.smallTopAppBarColors(containerColor = Color.White),
+            modifier = Modifier.testTag("preferencesTopBar"))
+      },
+      bottomBar = {
+        SaveButton(
+            onSaveClick = {
+              val newPreferences = Preferences(unitsSystem, weightUnit)
+              preferencesViewModel.updatePreferences(newPreferences)
+              navigationActions.goBack()
```

Comment:
Maybe for the Ux would be better to print a toast "Changes succesful" or something like that so that he knows everything went well.

Reply:
Good idea !

Done.